Sming/third-party/.patches/esp-open-lwip.patch

```diff
@@ -130,19 +130,64 @@ index ff03b30..fde6567 100644
  #define LWIP_PLATFORM_DIAG(x)
  #define LWIP_PLATFORM_ASSERT(x)
 diff --git a/include/lwipopts.h b/include/lwipopts.h
-index eaa8dd6..dbacd90 100644
+index eaa8dd6..6568657 100644
 --- a/include/lwipopts.h
 +++ b/include/lwipopts.h
-@@ -953,7 +953,7 @@
+@@ -403,7 +403,7 @@
+  * PBUF_POOL_SIZE: the number of buffers in the pbuf pool. 
+  */
+ #ifndef PBUF_POOL_SIZE
+-#define PBUF_POOL_SIZE                  10
++#define PBUF_POOL_SIZE                  8
+ #endif
+ 
+ /*
+@@ -422,7 +422,7 @@
+  * ARP_TABLE_SIZE: Number of active MAC-IP address pairs cached.
+  */
+ #ifndef ARP_TABLE_SIZE
+-#define ARP_TABLE_SIZE                  10
++#define ARP_TABLE_SIZE                  8
+ #endif
+ 
+ /**
+@@ -447,7 +447,7 @@
+  * Also notice that this slows down input processing of every IP packet!
+  */
+ #ifndef ETHARP_TRUST_IP_MAC
+-#define ETHARP_TRUST_IP_MAC             1
++#define ETHARP_TRUST_IP_MAC             0
```

Comment:
It will slow down initially to get ARP done but then it will be in the table. If trust==1 every incoming ip packet the source IP is searched in the ARP table and if not found it is appended in the arp cache.

```
#if ETHARP_TRUST_IP_MAC
      /* update ARP table */
      etharp_ip_input(netif, p);
#endif /* ETHARP_TRUST_IP_MAC */
```

in /sming/Sming/third-party/esp-open-lwip/lwip/netif/etharp.c